examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py

```diff
@@ -1505,7 +1561,8 @@ def load_model_hook(models, input_dir):
             models = [unet_]
             if args.train_text_encoder:
                 models.extend([text_encoder_one_, text_encoder_two_])
-            cast_training_params(models)
+                # only upcast trainable parameters (LoRA) into fp32
+                cast_training_params(models)
```

Comment:
unless you're using the AnyPrecision optimiser or the adamw-bf16 optimiser this shouldn't be indented. and in fact for AnyPrecision or AdamW-BF16 we should not run this at all. it should run whether or not the TE is trained.